functions/api/branch.js

```diff
@@ -164,7 +180,26 @@ router.put('/', isAdminOrSuperAdmin, async (request, response, next) => {
     }
     const oldData = branch.data()
     let newData = request.body
+    const branchName = newData[constants.NAME].toLocaleLowerCase()
+    if (newData[constants.IS_HEAD_OFFICE] === true) {
+        // Check whether there is a branch which is a head office
+        const headOfficeSnapshot = await db.collection(constants.BRANCHES)
+            .where(constants.IS_HEAD_OFFICE, '==', true)
+            .get()
+        if (headOfficeSnapshot.size > 0) {
+            var headOffice = ""
+            headOfficeSnapshot.forEach(branch => {
+                headOffice = branch.data()[constants.NAME]
+            })
```

Comment:
wont it make sense to have this check of isHeadBranchExists as separate function ?

Reply:
Done